src/main/scala/com/mozilla/telemetry/sinks/HTTPSink.scala

```diff
@@ -5,8 +5,24 @@ package com.mozilla.telemetry.streaming.sinks
 
 import scalaj.http.{Http, HttpRequest}
 import org.apache.spark.sql.ForeachWriter
+import scala.annotation.tailrec
+import scala.util.{Try, Success, Failure}
+
+class HttpSink[String](url: String, parameters: Map[String, String], maxAttempts: Int = 5, defaultDelay: Int = 500, connectionTimeout: Int = 2000)
+  extends ForeachWriter[String] {
+
+  // codes to retry a request on - allow retries on timeouts
+  // docs: https://amplitude.zendesk.com/hc/en-us/articles/204771828#http-status-codes-retrying-failed-requests
+
+  val TimeoutPseudoCode = -1
+  val RetryCodes = TimeoutPseudoCode :: 429 :: 500 :: 502 :: 503 :: 504 :: Nil
+  val OK = 200
+
+  // timeouts in ms
+  val ReadTimeout = 5000
+
+  @transient lazy val log = org.apache.log4j.LogManager.getLogger("HttpSinkLogger")
 
-class HttpSink[String](url: String, parameters: Map[String, String]) extends ForeachWriter[String] {
   def close(errorOrNull: Throwable): Unit = {}
```

Comment:
How about logging errors here?
We're logging some failed requests @L60, we might lose some error information here.